checkov/sca_package/scanner.py

```diff
@@ -24,7 +24,9 @@
 class Scanner:
     def __init__(self, pbar: ProgressBar = None, root_folder: str | Path | None = None) -> None:
         self._base_url = bc_integration.api_url
-        self.pbar = pbar
+        self.pbar = pbar or ProgressBar('')
+        if not pbar:
+            self.pbar.turn_off_progress_bar()
```

Comment:
```suggestion
        self.pbar = pbar
        if not self.pbar:
            self.pbar = ProgressBar('')
            self.pbar.turn_off_progress_bar()
```
did you mean this? 😄 otherwise it will never be `None`

Reply:
actually `self.pbar` will never be but `pbar` will.
and i like more your way

Reply:
Isn't that equal?
if pbar is `None` then self.pbar will be initiated with `ProgressBar('')` and so the condition will pass (the condition is on pbar, not self.pbar)